spec/condition_spec.rb

```diff
@@ -135,14 +135,333 @@
+  end
+
+  describe 'null handling' do
+    before(:context) do
+      @exact_browser_condition = {'name' => 'browser_type', 'match' => 'exact', 'type' => 'custom_attribute', 'value' => 'firefox'}
+      @exact_device_condition =  {'name' => 'device', 'match' => 'exact', 'type' => 'custom_attribute', 'value' => 'iphone'}
+      @exact_location_condition = {'name' => 'location', 'match' => 'exact', 'type' => 'custom_attribute', 'value' => 'san francisco'}
+    end
+    describe 'and evaluation' do
+      it 'should return null when all operands evaluate to null' do
+        user_attributes = {
+          'browser_type' => 4.5,
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['and', @exact_browser_condition, @exact_location_condition])).to eq(nil)
+      end
+
+      it 'should return null when operands evaluate to trues and nulls' do
+        user_attributes = {
+          'browser_type' => 'firefox',
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['and', @exact_browser_condition, @exact_location_condition])).to eq(nil)
+      end
+
+      it 'should return false when operands evaluate to falses and nulls' do
+        user_attributes = {
+          'browser_type' => 'chrome',
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['and', @exact_browser_condition, @exact_location_condition])).to be false
+      end
+
+      it 'should return false when operands evaluate to trues, falses, and nulls' do
+        user_attributes = {
+          'browser_type' => 'firefox',
+          'device' => 'android',
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['and', @exact_browser_condition, @exact_device_condition, @exact_location_condition])).to be false
+      end
+    end
+
+    describe 'or evaluation' do
+      it 'should return null when all operands evaluate to null' do
+        user_attributes = {
+          'browser_type' => 4.5,
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['or', @exact_browser_condition, @exact_location_condition])).to eq(nil)
+      end
+
+      it 'should return true when operands evaluate to trues and nulls' do
+        user_attributes = {
+          'browser_type' => 'firefox',
+          'location' => false
+        }
+        condition_evaluator = Optimizely::ConditionEvaluator.new(user_attributes)
+        expect(condition_evaluator.evaluate(['or', @exact_browser_condition, @exact_location_condition])).to be true
```

Comment:
Place this in a way that first condition evaluates to null and 2nd evaluates to True. 
It should still return True